### DISCUSSION TITLE
Adding host rules for npm private package with process.env variable.

### DISCUSSION THREAD

**Discussion by LinsonAlfred** [renovatebot/renovate#10935] [Request Help]
Hi There, 

My renovate.json looks like as given below. 
```
{
  "extends": ["config:base"],
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "token": process.env.MY_PAT
    }
  ],
  "npmrc": "@organizationName:registry=https://npm.pkg.github.com/"
}
```

How can i use  `"token": process.env.MY_PAT` this? 

After searching issues and discussions in `renovate` repo, I came to understand that I should use config.js instead of `renovate.json`. 

Right now I have placed renovate.json at the root folder of the repository. So if I am moving to config.js then just need to rename the renovate.json to config.js and place the code as below? 

```
module.exports = {
  "extends": ["config:base"],
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "token": process.env.MY_PAT
    }
  ],
  "npmrc": "@organizationName:registry=https://npm.pkg.github.com/"
}
```

My other question is if I am planning to use renovate.json, then in github workflow 

```
- name: NPM Install
        run: |
          make node/install
        env:
          NPM_TOKEN: ${{env.MY_PAT}}
          RENOVATE_TOKEN:${{env.MY_PAT}}

```

and my renovate.json will be look like this. 

```
{
  "extends": ["config:base"],
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm"
    }
  ],
  "npmrc": "@organizationName:registry=https://npm.pkg.github.com/"
}
```

Will this work?

Which one you suggest? 

1. Moving to config.js
2. Add env to workflow

**Comment by viceice** [comment]
- The `config.js` is the bot / admin config and needs to be in the runner repo.
- In repo config you can only use json files, so no `process.env`
- You need to configure the host rule in the renovate runner repo, if you like to use env vars.

---

You can use [secrets](https://docs.renovatebot.com/self-hosted-configuration/#secrets) to pass the token from bot to repo config.

**config.js**
```js
module.exports = {
  secrets: {
    NPM_TOKEN: process.env.NPM_TOKEN,
  },
};

```

**renovate.json**
```json
{
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "token": "{{ secrets.NPM_TOKEN}}"
    }
  ]
}
```

**Reply by LinsonAlfred** [reply]
@viceice Thank you for your response.
So, If I am using GithubApp installation https://docs.renovatebot.com/install-github-app/, then how should i do these changes?

**Reply by viceice** [reply]
checkout https://docs.renovatebot.com/private-modules/#encrypting-secrets

**renovate.json**
```json
{
  "hostRules": [
    {
      "matchHost": "https://npm.pkg.github.com/",
      "hostType": "npm",
      "encrypted": {
          "token": "XXXXXX"
      }
    }
  ]
}
